src/ray/raylet/local_object_manager.cc

```diff
@@ -343,46 +343,23 @@ void LocalObjectManager::AddSpilledUrls(
   }
 }
 
+std::string LocalObjectManager::GetSpilledObjectURL(const ObjectID &object_id) {
+  auto entry = spilled_objects_url_.find(object_id);
+  if (entry != spilled_objects_url_.end()) {
+    return entry->second;
+  } else {
+    return "";
+  }
+}
+
 void LocalObjectManager::AsyncRestoreSpilledObject(
-    const ObjectID &object_id, const std::string &object_url, const NodeID &node_id,
+    const ObjectID &object_id, const std::string &object_url,
     std::function<void(const ray::Status &)> callback) {
   if (objects_pending_restore_.count(object_id) > 0) {
     // If the same object is restoring, we dedup here.
     return;
   }
 
-  if (is_external_storage_type_fs_ && node_id != self_node_id_) {
```

Comment:
Do we still need `is_external_storage_type_fs_`?

Reply:
Nope!

Reply:
Ah, we still need it for a couple things like setting Node to Nil vs not nil, so we can't entirely remove it.